datastore/src/json_file.rs

```diff
@@ -272,34 +272,42 @@ mod tests {
     use futures::{Future, Stream};
     use tempfile::NamedTempFile;
     use {Filter, FilterOp, FilterTy, Order, Query};
+    use std::path::PathBuf;
+
+    fn gen_random_path() -> PathBuf {
+        let temp_file = NamedTempFile::new().unwrap();
+        temp_file.path().to_path_buf()
```

Comment:
This means that the path will not be destroyed at the end of the test.
Please just manually copy these lines in every test instead of using a separate function.